core/src/main/java/org/elasticsearch/common/blobstore/url/URLBlobStore.java

```diff
@@ -119,10 +119,10 @@ public void close() {
      * @return Base URL + path
      */
     private URL buildPath(BlobPath path) throws MalformedURLException {
-        String[] paths = path.toArray();
-        if (paths.length == 0) {
+        if (path.isEmpty) {
```

Comment:
@dadoonet I added isEmpty() public method to BlobPath class in first [commit](https://github.com/alexVengrovsk/elasticsearch/commit/6898f877e0b0c43bdddd600cbc0c1bdf88df85cd) of this PR